lib/beetle/dead_lettering.rb

```diff
@@ -10,54 +10,64 @@ def initialize(config)
     end
 
     def bind_dead_letter_queues!(channel, servers, target_queue, creation_keys = {})
-      return unless @config.dead_lettering_enabled?
-
-      dead_letter_queue_name = dead_letter_queue_name(target_queue)
-
-      logger.debug("Beetle: creating dead letter queue #{dead_letter_queue_name} with opts: #{creation_keys.inspect}")
-      channel.queue(dead_letter_queue_name, creation_keys)
-
-      logger.debug("Beetle: setting #{dead_letter_queue_name} as dead letter queue of #{target_queue} on all servers")
-      set_dead_letter_policies!(servers, target_queue)
+      if @config.dead_lettering_enabled?
+        dead_letter_queue_name = dead_letter_queue_name(target_queue)
+        logger.debug("Beetle: creating dead letter queue #{dead_letter_queue_name} with opts: #{creation_keys.inspect}")
+        channel.queue(dead_letter_queue_name, creation_keys)
+      end
 
-      logger.debug("Beetle: setting #{target_queue} as dead letter queue of #{dead_letter_queue_name} on all servers")
-      set_dead_letter_policies!(
-        servers,
-        dead_letter_queue_name,
-        :message_ttl => @config.dead_lettering_msg_ttl,
-        :routing_key => target_queue
-      )
+      if @config.dead_lettering_enabled?
```

Comment:
A bit strange to see the same `if @config.dead_lettering_enabled?` again, i.e. the following `logger.debug` line could be moved up.

Reply:
That is true, but I wanted the log statement line grouped with the action that is executed. They belong together.

Reply:
Originally I had used "logger.debug ... if ...", but that looked weird, IMO.

Reply:
I see. Fine 😉

Reply:
We could move it into set_queue_policies, but that's also kind of weird.